lib/session.js

```diff
@@ -201,22 +201,23 @@ Session.prototype.begin = function(sessionPolicy) {
 };
 
 Session.prototype.createLink = function(linkPolicy) {
-  var policy = u.deepMerge(linkPolicy);
+  var policy = u.deepMerge(linkPolicy),
+      attachOptions = policy.attach || {};
 
-  policy.options.handle = this._nextHandle();
-  if (typeof policy.options.name === 'function')
-    policy.options.name = policy.options.name();
+  attachOptions.handle = this._nextHandle();
+  if (typeof attachOptions.name === 'function')
+    attachOptions.name = attachOptions.name();
```

Comment:
![It's magic!](http://1.bp.blogspot.com/-hlFCCSnGtWQ/UFnvgzXpuhI/AAAAAAAAAfE/aVu6BaRCmiA/s320/its_magic.jpg)